test/internet/test-uv-threadpool-schedule.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+// test to validate massive dns lookups do not block filesytem I/O
+// (or any fast I/O).Prior to https://github.com/libuv/libuv/pull/1845
+// few back-to-back dns lookups were sufficient to engage libuv
+// threadpool workers in a blocking manner, throttling other work items
+// that need pool resources. Start slow and fast I/Os together, make sure
+// fast I/O can complete in at least in 1/100th of time for slow I/O.
+// TEST TIME TO COMPLETION: ~5 seconds.
+
+const common = require('../common');
+const dns = require('dns');
+const fs = require('fs');
+const assert = require('assert');
+
+const start = Date.now();
+
+const slowIOmax = 100;
+let slowIOcount = 0;
+let fastIOdone = false;
+let slowIOend, fastIOend;
+
+for (let i = 0; i < slowIOmax; i++) {
+  // We need to refresh the domain string everytime,
+  // otherwise the TCP stack that cache the previous lookup
+  // returns result from memory, breaking all our Math.
+  dns.lookup(`${randomDomain()}.com`, {}, common.mustCall(() => {
```

Comment:
Nit: The callback function can be moved out and used with `common.mustCall(..., slowIOmax)`.

Reply:
I moved the callback body out of the loop. However, issue with attaching expected call count to the `common.mustCall` is: the loop executes 100 times, and each time when we invoke `common.mustCall` with `slowIOmax`, the total expectation becomes 10K. So I left it as is (default:1). Hope this is fine with you?

Reply:
I was thinking about something like

```js
const onResolve = common.mustCall(..., slowIOmax);
...
dns.lookup(..., onResolve);
```

But your current change is intuitive. Let's go with that.